Produce correct HTML for EXPORT_ES6

When trying to output a `.html` with `-s MODULARIZE=1 -s EXPORT_ES6=1`, I'm getting the following error:

```
$ emcc -s MODULARIZE=1 -s EXPORT_ES6=1 -O -g2 -o dist/fibonacci.html fibonacci.c
emcc: error: Due to collision in variable name "Module", the shell file "/home/rreverser/emsdk/upstream/emscripten/src/shell.html" is not compatible with build options "-s MODULARIZE=1 -s EXPORT_NAME=Module". Either provide your own shell file, change the name of the export to something else to avoid the name collision. (see https://github.com/emscripten-core/emscripten/issues/7950 for details)
```

The error is talking about a clash between `-s MODULARIZE` and `-s EXPORT_NAME=Module`. But I'm not using `EXPORT_NAME` - surely, with `EXPORT_ES6` the output name doesn't matter that much (it's not publicly accessible anyway) and Emscripten could take care of defining one that wouldn't collide with HTML that is also produced by Emscripten.

Moreover, if I try to specify `EXPORT_NAME` myself, I'm now getting a different error:

```
$ emcc -s MODULARIZE=1 -s EXPORT_ES6=1 -s EXPORT_NAME=fibonacci -O -g2 -o dist/fibonacci.html fibonacci.c
emcc: error: Customizing EXPORT_NAME requires that the HTML be customized to use that name (see https://github.com/emscripten-core/emscripten/issues/10086)
make: *** [Makefile:63: dist/fibonacci-no-debug.html] Error 1
```

This seems like a frustrating experience, and, given that both clashing names are internal, it feels like something that Emscripten should take care of itself.

Perhaps the root issue is that autogenerated HTML currently doesn't have support for `EXPORT_ES6` at all? In that case, the fix would be adding such support and generating `import` statement instead of a script include.

--- a/emcc.py
+++ b/emcc.py
@@ -34,7 +34,6 @@ import sys
 import time
 from enum import Enum, unique, auto
 from subprocess import PIPE
-from urllib.parse import quote
 
 
 import emscripten
@@ -2009,7 +2008,8 @@ def phase_linker_setup(options, state, newargs, settings_map):
     # Require explicit -lfoo.js flags to link with JS libraries.
     settings.AUTO_JS_LIBRARIES = 0
 
-  if settings.MODULARIZE and settings.EXPORT_NAME == 'Module' and options.oformat == OFormat.HTML and \
+  if settings.MODULARIZE and not (settings.EXPORT_ES6 and not settings.SINGLE_FILE) and \
+     settings.EXPORT_NAME == 'Module' and options.oformat == OFormat.HTML and \
      (options.shell_path == shared.path_from_root('src', 'shell.html') or options.shell_path == shared.path_from_root('src', 'shell_minimal.html')):
     exit_with_error('Due to collision in variable name "Module", the shell file "' + options.shell_path + '" is not compatible with build options "-s MODULARIZE=1 -s EXPORT_NAME=Module". Either provide your own shell file, change the name of the export to something else to avoid the name collision. (see https://github.com/emscripten-core/emscripten/issues/7950 for details)')
 
@@ -3493,18 +3493,33 @@ class ScriptSource:
     """Use this if you want to modify the script and need it to be inline."""
     if self.src is None:
       return
-    self.inline = '''
-          var script = document.createElement('script');
-          script.src = "%s";
-          document.body.appendChild(script);
-''' % self.src
+    if settings.EXPORT_ES6:
+      template = '''
+        import("./%s").then(exports => exports.default(Module))
+      '''
+    else:
+      template = '''
+            var script = document.createElement('script');
+            script.src = "%s";
+            document.body.appendChild(script);
+      '''
+    self.inline = template % self.src
     self.src = None
 
   def replacement(self):
     """Returns the script tag to replace the {{{ SCRIPT }}} tag in the target"""
     assert (self.src or self.inline) and not (self.src and self.inline)
     if self.src:
-      return '<script async type="text/javascript" src="%s"></script>' % quote(self.src)
+      if settings.EXPORT_ES6:
+        template = '''
+        <script type="module">
+          import initModule from "./%s";
+          initModule(Module);
+        </script>
+        '''
+      else:
+        template = '<script async type="text/javascript" src="%s"></script>'
+      return template % self.src
     else:
       return '<script>\n%s\n</script>' % self.inline
 

--- a/emcc.py
+++ b/emcc.py
@@ -34,6 +34,7 @@ import sys
 import time
 from enum import Enum, unique, auto
 from subprocess import PIPE
+from urllib.parse import quote
 
 
 import emscripten
@@ -3493,33 +3494,33 @@ class ScriptSource:
     """Use this if you want to modify the script and need it to be inline."""
     if self.src is None:
       return
+    quoted_src = quote(self.src)
     if settings.EXPORT_ES6:
-      template = '''
-        import("./%s").then(exports => exports.default(Module))
+      self.inline = f'''
+        import("./{quoted_src}").then(exports => exports.default(Module))
       '''
     else:
-      template = '''
+      self.inline = f'''
             var script = document.createElement('script');
-            script.src = "%s";
+            script.src = "{quoted_src}";
             document.body.appendChild(script);
       '''
-    self.inline = template % self.src
     self.src = None
 
   def replacement(self):
     """Returns the script tag to replace the {{{ SCRIPT }}} tag in the target"""
     assert (self.src or self.inline) and not (self.src and self.inline)
     if self.src:
+      quoted_src = quote(self.src)
       if settings.EXPORT_ES6:
-        template = '''
+        return f'''
         <script type="module">
-          import initModule from "./%s";
+          import initModule from "./{quoted_src}";
           initModule(Module);
         </script>
         '''
       else:
-        template = '<script async type="text/javascript" src="%s"></script>'
-      return template % self.src
+        return f'<script async type="text/javascript" src="{quoted_src}"></script>'
     else:
       return '<script>\n%s\n</script>' % self.inline
 

--- a/tests/test_browser.py
+++ b/tests/test_browser.py
@@ -172,6 +172,9 @@ class browser(BrowserCore):
     self.btest('hello_world_sdl.cpp', reference='htmltest.png', args=['-lSDL', '-lGL'])
     self.btest('hello_world_sdl.cpp', reference='htmltest.png', args=['-s', 'USE_SDL', '-lGL']) # is the default anyhow
 
+  def test_sdl1_es6(self):
+    self.btest('hello_world_sdl.cpp', reference='htmltest.png', args=['-s', 'USE_SDL', '-lGL', '-s', 'EXPORT_ES6'])
+
   # Deliberately named as test_zzz_* to make this test the last one
   # as this test may take the focus away from the main test window
   # by opening a new window and possibly not closing it.

